dotCMS/src/main/java/com/dotcms/rest/api/v1/site/SiteHelper.java

```diff
@@ -98,6 +128,55 @@ public Host getSite(User user, String siteId) throws DotSecurityException, DotDa
 		return site;
 	}
 
+	/**
+	 * Return a site by user and site id, respect front roles = false
+	 *
+	 * @param user User to filter the host to return
+	 * @param siteId Id to filter the host to return
+	 * @return host that the given user has permissions and with id equal to hostId, if any exists then return null
+	 * @throws DotSecurityException if one is thrown when the sites are search
+	 * @throws DotDataException if one is thrown when the sites are search
+	 */
+	public Host getSiteNoFrontEndRoles(final User user, final String siteId) throws DotSecurityException, DotDataException {
+
+		final Host site = this.hostAPI.find(siteId, user, Boolean.FALSE);
+		return site;
+	}
+
+	/**
+	 * Publish a host
+	 * @param host {@link Host}
+	 * @param user {@link User}
+	 * @param respectAnonPerms {@link Boolean}
+	 * @return Host
+	 * @throws DotContentletStateException
+	 * @throws DotDataException
+	 * @throws DotSecurityException
+	 */
+	@WrapInTransaction
+	public Host publish(final Host host, final User user, final boolean respectAnonPerms) throws DotContentletStateException, DotDataException, DotSecurityException {
+
+		this.hostAPI.publish(host, user, respectAnonPerms);
+		return host;
+	}
+
+	/**
+	 * Unpublish a host
+	 * @param host {@link Host}
+	 * @param user {@link User}
+	 * @param respectAnonPerms {@link Boolean}
+	 * @return Host
+	 * @throws DotContentletStateException
+	 * @throws DotDataException
+	 * @throws DotSecurityException
+	 */
+	@WrapInTransaction
+	public Host unpublish(final Host host, final User user, final boolean respectAnonPerms) throws DotContentletStateException, DotDataException, DotSecurityException {
+
+		this.hostAPI.unpublish(host, user, respectAnonPerms);
+		return host;
+	}
+
 	/**
 	 * Determines what site is to be marked as "selected" by a user. If the
```

Comment:
do The helper methods need the @WrapInTransaction?, this anotation should not be in the API methods?